magicctapipe/scripts/lst1_magic/lst1_magic_event_coincidence.py

```diff
@@ -182,6 +183,121 @@ def event_coincidence(input_file_lst, input_dir_magic, output_dir, config):
         df_magic["timestamp"] = timestamps_magic.to_value("s")
         df_magic.drop(["time_sec", "time_nanosec"], axis=1, inplace=True)
 
+        # Pre offset search is performed to define the offset scan region.
+        # First, N events are extracted from largerst intensity events for LST and
+        # MAGIC. Then, it counts a number of coincident events within a defined
+        # window after shifting all possible combination (N x N) of time offsets.
+        if pre_offset_search:
+            logger.info(
+                "\nPre offset search using large-intensity shower events is ongoing..."
+            )
+
+            logger.info(
+                f"\nExtracting the {tel_name} events taken when LST-1 observed for pre offset search..."
+            )
+
+            time_lolim = timestamps_lst[0] - window_half_width
+            time_uplim = timestamps_lst[-1] + window_half_width
+
+            cond_lolim = timestamps_magic >= time_lolim
+            cond_uplim = timestamps_magic <= time_uplim
+
+            mask_lst_obs_window = np.logical_and(cond_lolim, cond_uplim)
+            n_events_magic = np.count_nonzero(mask_lst_obs_window)
+
+            if n_events_magic == 0:
+                logger.info(f"--> No {tel_name} events are found. Skipping...")
+                continue
```

Comment:
here we continue to the next tel_id, but I can imagine that if for M1 there are no events fulfilling the condition above, then also M2 should not fulfill, right? Probably if for one of the telescope the number of events in the LST window is 0, we should exit from the script with some exception and write an informative message

Reply:
On 2nd thought, since the coincidence script is run over MAGIC DL1 files, it means the loop will be on 1 telescope only, either M1 or M2, so we definitely need to exit if no events are found

Reply:
Here I simply copy&pasted the existing lines, so maybe @YoshikiOhtani can comment on that. I saw sometimes this warning message (LST run 12273 taken on March 18 2023) and only LST-M2 events are found. I will check later...

https://github.com/cta-observatory/magic-cta-pipe/blob/b9f82d982143008bd053690dc32c9de89db37a32/magicctapipe/scripts/lst1_magic/lst1_magic_event_coincidence.py#L197-L199

Reply:
Yes, I also saw that the coincidence happens in only one of the combinations, even if I input the M1 and M2 datasets to the script. Maybe this is due to the case that `calib_to_dl1` process fails for some subrun files of either M1 and M2, but the counterparts are still used in the coincidence search. Anyway, this is out of the scope of this pull request so I think we should open another issue page.

Reply:
Regarding the second thought, we can input the M1 and M2 datasets to the script simultaneously with `--input-dir-magic`, and in that case the loop will be on the two telescopes.